### PR TITLE
Add ordering detail of xml layout merging files

### PR DESCRIPTION
**Pull request by speedy008**
## Purpose of this pull request

This pull request (PR) contains ordering detail of XML layout merging files

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.3/frontend-dev-guide/layouts/xml-instructions.html
- https://devdocs.magento.com/guides/v2.2/frontend-dev-guide/layouts/xml-instructions.html

whatsnew
Added a [new section](https://devdocs.magento.com/guides/v2.3/frontend-dev-guide/layouts/xml-instructions.html#xml_Layout_Merging) about the order of XML layout merging

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.2/frontend-dev-guide/layouts/xml-instructions.md [hunk 1]
```diff
@@ -298,6 +300,23 @@ Example:
...
+**Module Files**
+Sorted according to their module priority (array index of module's position in app/etc/config.php). If their priority are equal, they are sorted according to their alphabetical priority.
+
+- Module base file loaded
+
+- Module Area files loaded
+
+**Theme files**
+
+- Layout file loaded
```
Comment (rogyar) [inline guides/v2.2/frontend-dev-guide/layouts/xml-instructions.md]: Hi @speedy008. Thank you for the provided description. 
We need to make the explanation more transparent for the end-user. So we have something like the following list
```suggestion
- Module base layout file loaded
- Module area layout file loaded
- Theme layout file loaded
...
```

Also, there's one explanation missing regarding theme layouts from the inheritance perspective. Usually a theme extends another theme, so layouts from a parent theme also should be taken into account.

Comment (rogyar) [reply guides/v2.2/frontend-dev-guide/layouts/xml-instructions.md]: Hi @speedy008. By the three dots in my example, I meant that the other contents of the list should follow the example, so we don't need the three dots in the list literally. 

To clarify my request, what needs to be fixed from your initial commit:
- Specify the layout source (i.e. module layout, theme layout)
- Add information about a parent theme layout loading order (it was missing in your initial commit)

Thank you

Comment (speedy008) [reply guides/v2.2/frontend-dev-guide/layouts/xml-instructions.md]: I have added path information please review.

Comment (rogyar) [reply guides/v2.2/frontend-dev-guide/layouts/xml-instructions.md]: The information about loading order of a child -> parent theme is missing. It's a very important part of the theme layout merging flow

Comment (shrielenee) [reply guides/v2.2/frontend-dev-guide/layouts/xml-instructions.md]: @speedy008 Can you confirm whether the loading order of a child -> parent theme is needed, and make those changes if necessary?

Comment (speedy008) [reply guides/v2.2/frontend-dev-guide/layouts/xml-instructions.md]: @shrielenee The information I have added is include parent theme information as well.

**Comment by rogyar** [comment]
Hi @speedy008. I still don't see my mention about themes inheritance is added to the description. Please, add this clarification. Example:
> Theme files are loaded (basic, parent and current themes). The younger (current) the theme has the highest priority in the extending and overriding layout files.

**Comment by osrecio** [comment]
I'm Closing this PR in for inactivity. @speedy008 Feel free to reopen it

**Pull request closed without merging**